src/Sunset/EventListener/AddSunsetHeaderListener.php

```diff
@@ -0,0 +1,40 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+/**
+ * Created by PhpStorm.
+ * User: blanky
+ * Date: 10/10/18
+ * Time: 9:06 PM.
+ */
+
+namespace ApiPlatform\Core\Sunset\EventListener;
+
+use Symfony\Component\HttpKernel\Event\FilterResponseEvent;
+
+class AddSunsetHeaderListener
+{
+    private $sunset;
+
+    public function __construct(string $sunset = null)
```

Comment:
The listener should be stateless. You should detect the current resource using `RequestAttributesExtractor` and call `ResourceMetadataFactory` instead.

Reply:
Made some changes. I am still not 100% confident this is how it should be done though. Do you think there should also be some sort of enforcing the Date format?